src/live.rs

```diff
@@ -776,6 +776,69 @@ fn copy_file_from_iso(iso: &mut IsoFs, file: &iso9660::File, output_path: &Path)
 mod tests {
     use super::*;
 
+    use std::io::copy;
+
+    use tempfile::tempfile;
+    use xz2::read::XzDecoder;
+
+    fn open_iso_file() -> File {
+        let iso_bytes: &[u8] = include_bytes!("../fixtures/embed-areas.iso.xz");
+        let mut decoder = XzDecoder::new(iso_bytes);
+        let mut iso_file = tempfile().unwrap();
+        copy(&mut decoder, &mut iso_file).unwrap();
+        iso_file
+    }
+
+    #[test]
+    fn test_ignition_embed_area() {
+        let mut iso_file = open_iso_file();
+        // normal read
+        let mut iso = IsoFs::from_file(iso_file.try_clone().unwrap()).unwrap();
+        let region = ignition_embed_area(&mut iso).unwrap();
+        assert_eq!(region.offset, 102400);
```

Comment:
Minor: would be nice if these offset numbers were actually calculated so it's easier to tell it's correct. Shouldn't be too hard with the IsoFs APIs.

Reply:
Sure, but then we're partially comparing the code itself.  It seemed better to manually generate the ground truth.